### PR TITLE
proposals implemented

### PR DESCRIPTION
**Pull request by evidentti**
This is a base of implementation. Some data is missing.

### CODE REVIEW

**Comment by evidentti** [comment]
wysiwyg icons are not showing correctly in Safari

**Comment by evidentti** [comment]
wysiwyg format buttons are out commented while solution for Safari is found.

File: app/directives/proposal/proposal.directive.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            self.endEditing = function() {
+                if ($scope.proposal.text !== self.editedText) {
+                    $scope.proposal.text = self.editedText;
+                    $scope.onPost({ proposal: $scope.proposal });
+                }
+                self.mode = PROP.MODE.OPEN;
+            };
+
+            self.cancelEditing = function() {
+                self.mode = (self.editedText && self.editedText.lenght) ? PROP.MODE.OPEN : PROP.MODE.COLLAPSED;
```
Comment (tkenakka) [inline app/directives/proposal/proposal.directive.js]: Typo in `self.editedText.lenght`

File: app/directives/proposal/proposal.directive.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            self.open = function() {
+                self.mode = PROP.MODE.OPEN;
+            };
+
+            self.collapse = function() {
+                self.mode = PROP.MODE.COLLAPSED;
+            };
+
+            self.typeText = function(value) {
+                for (var index = 0; index < PROPS.TYPE.length; index++) {
```
Comment (tkenakka) [inline app/directives/proposal/proposal.directive.js]: Here one could consider using the rootScope util function `objWithVal` which is already used by
e.g. opensignreqs.directive.js and signitem.ctrl.js

File: app/directives/proposals/proposals.directive.html [hunk 3]
```diff
@@ -1,3 +1,39 @@
 <div>
-    {{ctrl.proposals}}
+    <div class="col-xs-12 prop-header">
+        <div class="prop-row">
```
Comment (tkenakka) [inline app/directives/proposals/proposals.directive.html:3]: Out of curiosity, why flex needed here and not for other rows? Didn't see big difference if it's omitted.

**Comment by tkenakka** [comment]
LGTM as a development version, but please fix the commented issues before merging.

File: app/directives/proposal/proposal.directive.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            self.isPublished = false;
+            self.isUser = false;
+            self.editedText = "";
+            self.smallEditor = {
+                'menu': [
+                    ['bold', 'italic', 'underline', 'strikethrough', 'subscript', 'superscript']
+                ]
+            };
+            self.editor = {
+                'menu': [
```
Comment (tkenakka) [inline app/directives/proposal/proposal.directive.js:45]: To be checked if proposal text formatting is needed at all, presumably not.

File: app/directives/proposals/proposals.directive.html (hunk 3 above)
Comment (evidentti) [inline app/directives/proposals/proposals.directive.html:3]: This is header and flex is needed to set text and button at same line. Influence can be seen better in small devices. Without flex button is on the top of parent space.

File: app/directives/proposal/proposal.directive.js (hunk 4 above)
Comment (evidentti) [inline app/directives/proposal/proposal.directive.js:45]: to be discussed.

**Pull request merged**